### PR TITLE
⚡ Enhance testing scripts

### PR DESCRIPTION
**Pull request by athak**
```
- Cleanup test init script
- Refactor to use a dedicated directory for Vault binaries
- Add test cleanup script
- Add a flag to optionally keep test vault instance running

PR #6
Close [ch1115]
```

### CODE REVIEW

File: test-files/initVaultDev.sh [hunk 1]
```diff
@@ -15,37 +31,49 @@ case "$(uname -s)" in
...
+  # Create policies
+  ${VAULT_BIN} policy write VaultDevAdmin "${SCRIPT_DIR}/VaultPolicy.hcl"
+  ${VAULT_BIN} policy write VaultNoKV "${SCRIPT_DIR}/NoKVVaultPolicy.hcl"
+  # Create AppRoles
+  ${VAULT_BIN} auth enable approle
+  ${VAULT_BIN} write auth/approle/role/my-role policies=VaultDevAdmin token_num_uses=100 token_ttl=10s token_max_ttl=300m secret_id_num_uses=40
+  ${VAULT_BIN} write auth/approle/role/no-kv policies=VaultNoKV token_num_uses=2 token_ttl=30m token_max_ttl=300m secret_id_num_uses=40
+} >> "${VAULT_LOG}"
+
+#unset VAULT_TOKEN
```
Comment (jambonrose) [inline test-files/initVaultDev.sh]: I see that we're removing the token in the clean up script and believe we can remove this.

**Review by jambonrose — CHANGES_REQUESTED** [review]
This looks good! I would like our indentation to follow the original author's style (some of the indentation in this code is not consistent) and I also have a an in-line cleanup request.

**Comment by athak** [comment]
> This looks good! I would like our indentation to follow the original author's style (some of the indentation in this code is not consistent) and I also have a an in-line cleanup request.

By indentation you mean using tabs? Or size? I think the script was already intented at 2...

File: test-files/initVaultDev.sh (hunk 1 above)
Comment (athak) [reply test-files/initVaultDev.sh]: I commented it for testing before adding the cleanup script and forgot to remove it. Done!

File: integration_test.go [hunk 2]
```diff
@@ -34,7 +47,8 @@ func prepareVault() {
 	if err != nil {
 		log.Fatalf("Error in initVaultDev.sh %v", err)
 	}
-	cmd := exec.Command("./vault", "read", "-field=role_id", "auth/approle/role/my-role/role-id")
+    vaultBin := "./test-files/vault/vault" + vaultVersion
+	cmd := exec.Command(vaultBin, "read", "-field=role_id", "auth/approle/role/my-role/role-id")
```
Comment (jambonrose) [inline integration_test.go:51]: ↑ Here is an example of inconsistent indentation.

File: integration_test.go [hunk 3]
```diff
@@ -20,12 +21,24 @@ func TestMain(m *testing.M) {
...
 	fmt.Println("TestMain: Preparing Vault server")
 	prepareVault()
 	fmt.Println("TestMain: Vault Prepared; Running Tests")
 	ret := m.Run()
+	if !keepVault {
+	  fmt.Println("TestMain: Cleaning up")
+	  stopVault()
+	} else {
+	  fmt.Println("TestMain: Not cleaning up as requested")
+	}
```
Comment (jambonrose) [inline integration_test.go:41]: ↑ Consider the indents in this new block compared to the previous blocks.

File: integration_test.go (hunk 2 above)
Comment (athak) [reply integration_test.go:51]: Yeah...tabs vs spaces...on it

**Review by dericmiller — APPROVED** [review]
This looks like exactly what we need.  I confirm that this code seems sensible, and that the new augmentations to testing work as desired.

File: test-files/initVaultDev.sh [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
+declare -r VAULT_BIN="${VAULT_BIN_DIR}/vault${VAULT_VERSION}"
+declare -r VAULT_LOG_DIR="${SCRIPT_DIR}/logs"
+declare -r VAULT_LOG="${VAULT_LOG_DIR}/vault${VAULT_VERSION}.log"
+declare -r VAULT_PID="${SCRIPT_DIR}/vault.pid"
+
+# Check for leftover processes from previous run
+if [[ -f "${VAULT_PID}" ]]; then
+  echo "There is a pid file present from a previous run. Aborting!"
+  exit 1
+fi
```
Comment (jambonrose) [inline test-files/initVaultDev.sh:20]: This block is not consistently formatted compared to the block below.

Comment (athak) [reply test-files/initVaultDev.sh:20]: What do you mean by formatted?

Comment (jambonrose) [reply test-files/initVaultDev.sh:20]: The block from lines 40 to 43 is indented differently than line 18 and 19

**Review by jambonrose — APPROVED** [review]
To test this, I pulled Deric's open branch, merged this branch in, and ran tests repeatedly to recreate the error documented in PR #5. I confirm that (after killing a Vault process that had not terminated from a previous test run) I am able to run tests repeatedly and consistently.

**Pull request merged**